### PR TITLE
[Inductor] Fix error checking for scaled_mm lowering

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134765

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: torch/_inductor/kernel/mm_scaled.py [hunk 1]
```diff
@@ -207,10 +219,11 @@ def scaled_mm_options(  # type: ignore[no-untyped-def]
         sympy.gcd(sym_k, config.kwargs["BLOCK_K"]) == config.kwargs["BLOCK_K"]
     )
 
-    assert len(scale_a.get_size()) == len(
-        scale_b.get_size()
-    ), "Expect inverse scale_a and scale_b to be both scalars (tensor-wise scaling) or tensors (rowwise scaling)."
-
+    size_a, size_b = scale_a.get_size(), scale_b.get_size()
+    assert are_compatible_scales(size_a, size_b), (
+        f"Expect scale_a and scale_b to be either both scalars (including single-element tensors) "
```
Comment (Skylion007) [inline torch/_inductor/kernel/mm_scaled.py:224]: Probably shouldn't be an f-string, ruff will likely complain

File: torch/_inductor/kernel/mm_scaled.py [hunk 2]
```diff
@@ -192,6 +192,18 @@
 aten__fp8_mm = ExternKernelChoice(torch._scaled_mm, "at::_scaled_mm")
 
 
+def are_compatible_scales(size_a: List[int], size_b: List[int]) -> bool:
```
Comment (Skylion007) [inline torch/_inductor/kernel/mm_scaled.py:195]: Technically could be `collections.abc.Sized` ;)

File: torch/_inductor/kernel/mm_scaled.py (hunk 1 above)
Comment (drisspg) [reply torch/_inductor/kernel/mm_scaled.py:224]: ahhh yeah will change

**Pull request closed without merging**